crates/bevy_ecs/src/component.rs

```diff
@@ -274,6 +275,23 @@ impl ComponentHooks {
             .expect("Component id: {:?}, already has an on_insert hook")
     }
 
+    /// Register a [`ComponentHook`] that will be run when this component is replaced (with `.insert`) or removed.
```

Comment:
I think this doc should be clearer.
It makes it sound like this is equivalent to `ON_INSERT` OR `ON_REMOVE`. Is that the case?

Would the component queries during the `ON_REPLACE` be the previous version of the component (before .insert()) or the replaced version?

Also maybe the docs should mention why this is useful, as it's not completely obvious at first glance. It's to run any teardown logic when a component gets overwritten?